src/provider/provider_coarse.c

```diff
@@ -447,15 +458,25 @@ static block_t *free_block_merge_with_prev(struct ravl *all_blocks,
 // free_block_merge_with_next - merge the given free block
 // with the next one if both are unused and have continuous data.
 // Remove the merged block from the tree of free blocks.
-static block_t *free_block_merge_with_next(struct ravl *all_blocks,
-                                           struct ravl *free_blocks,
-                                           block_t *block) {
+static block_t *free_block_merge_with_next(
+    umf_memory_provider_handle_t upstream_memory_provider,
+    struct ravl *all_blocks, struct ravl *free_blocks, block_t *block) {
+
+    assert(all_blocks);
     assert(free_blocks);
     assert(block);
 
+    // check if blocks could be merged by the upstream provider
+    enum umf_result_t merge_success = UMF_RESULT_ERROR_UNKNOWN;
     if (block->next && block->next->used == false &&
```

Comment:
I think that a check for the usage of block (block->used == false) is missing.

Reply:
added assert